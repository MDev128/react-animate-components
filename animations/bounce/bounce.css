@keyframes bounce {
  from {
    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);
  }

  0% {
    opacity: 0;
    transform: scale3d(.3, .3, .3);
  }

  20% {
    transform: scale3d(1.1, 1.1, 1.1);
  }

  40% {
    transform: scale3d(.9, .9, .9);
  }

  60% {
    opacity: 1;
    transform: scale3d(1.1, 1.1, 1.1);
  }

  80% {
    transform: scale3d(.97, .97, .97);
  }

  to {
    opacity: 1;
    transform: scale3d(1, 1, 1);
  }
}

.bounce {
  animation-name: bounce
}
