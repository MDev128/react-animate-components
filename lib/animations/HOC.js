"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var HOC = function HOC(componentName, animationName) {
  var _class, _temp2;

  return _temp2 = _class = function (_Component) {
    _inherits(_class, _Component);

    function _class() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, _class);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = _class.__proto__ || Object.getPrototypeOf(_class)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
        styles: {}
      }, _this.componentDidMount = function () {
        var _this$props = _this.props,
            duration = _this$props.duration,
            timingFunction = _this$props.timingFunction,
            delay = _this$props.delay,
            direction = _this$props.direction,
            iterations = _this$props.iterations,
            backfaceVisible = _this$props.backfaceVisible,
            fillMode = _this$props.fillMode;


        _this.setState({
          styles: {
            display: "inline-block" /** <span> elements can't be animated with CSS. In order to animate them with 
                                    CSS we assign display property to them */
            , // Works but will be updated later.
            animation: animationName + " " + duration + " " + timingFunction + " " + delay + " " + fillMode + " " + direction + " " + iterations,
            backfaceVisibility: "" + backfaceVisible
          }
        });
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(_class, [{
      key: "render",


      // Using <span> instead of composedComponent because each component has a base component <span> only.
      value: function render() {
        return _react2.default.createElement(
          "span",
          { style: this.state.styles },
          this.props.children
        );
      }
    }]);

    return _class;
  }(_react.Component), _class.defaultProps = {
    duration: "1s",
    timingFunction: "ease",
    delay: "0s",
    direction: "normal",
    iterations: "1",
    backfaceVisible: "visible",
    fillMode: "none"
  }, _temp2;
};

exports.default = HOC;