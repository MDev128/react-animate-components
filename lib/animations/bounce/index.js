'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _styledBounce = require('./styledBounce');

var _HOC = require('../HOC');

var _HOC2 = _interopRequireDefault(_HOC);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Bounce animation components
var bounceObj = {
	Bounce: (0, _HOC2.default)('Bounce', _styledBounce.bounce),
	BounceDown: (0, _HOC2.default)('BounceDown', _styledBounce.bounceDown),
	BounceUp: (0, _HOC2.default)('BounceUp', _styledBounce.bounceUp),
	BounceLeft: (0, _HOC2.default)('BounceLeft', _styledBounce.bounceLeft),
	BounceRight: (0, _HOC2.default)('BounceRight', _styledBounce.bounceRight)
};

exports.default = bounceObj;