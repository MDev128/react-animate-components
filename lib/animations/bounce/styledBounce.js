'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bounceRight = exports.bounceLeft = exports.bounceUp = exports.bounceDown = exports.bounce = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: scale3d(.3, .3, .3);\n  }\n\n  20% {\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  40% {\n    transform: scale3d(.9, .9, .9);\n  }\n\n  60% {\n    opacity: 1;\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  80% {\n    transform: scale3d(.97, .97, .97);\n  }\n\n  to {\n    opacity: 1;\n    transform: scale3d(1, 1, 1);\n  }\n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: scale3d(.3, .3, .3);\n  }\n\n  20% {\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  40% {\n    transform: scale3d(.9, .9, .9);\n  }\n\n  60% {\n    opacity: 1;\n    transform: scale3d(1.1, 1.1, 1.1);\n  }\n\n  80% {\n    transform: scale3d(.97, .97, .97);\n  }\n\n  to {\n    opacity: 1;\n    transform: scale3d(1, 1, 1);\n  }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, -2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, 25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, -15px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, 5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  }\n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, -2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, 25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, -15px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, 5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  }\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, 2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, -25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, 10px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, -5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  } \n'], ['\n  from {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    transform: translate3d(0, 2000px, 0);\n  }\n\n  60% {\n    transform: translate3d(0, -25px, 0);\n  }\n\n  75% {\n    transform: translate3d(0, 10px, 0);\n  }\n\n  90% {\n    transform: translate3d(0, -5px, 0);\n  }\n\n  to {\n    transform: translate3d(0, 0, 0);\n  } \n']),
    _templateObject4 = _taggedTemplateLiteral(['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(-2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(-10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n'], ['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(-2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(-10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n']),
    _templateObject5 = _taggedTemplateLiteral(['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(-25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(-5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n'], ['\n  from, 60% 75% 90%, to {\n    animation-timing-function: cubic-bezier(0.200, 0.620, 0.340, 1.000);\n  }\n\n  0% {\n    opacity: 0;\n    transform: translate3d(2000px, 0, 0);\n  }\n\n  60% {\n    opacity: 1;\n    transform: translate3d(-25px, 0, 0);\n  }\n\n  75% {\n    transform: translate3d(10px, 0, 0);\n  }\n\n  90% {\n    transform: translate3d(-5px, 0, 0);\n  }\n\n  to {\n    transform: none;\n  } \n']);

var _styledComponents = require('styled-components');

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var bounce = (0, _styledComponents.keyframes)(_templateObject);

var bounceDown = (0, _styledComponents.keyframes)(_templateObject2);

var bounceUp = (0, _styledComponents.keyframes)(_templateObject3);

var bounceLeft = (0, _styledComponents.keyframes)(_templateObject4);

var bounceRight = (0, _styledComponents.keyframes)(_templateObject5);

exports.bounce = bounce;
exports.bounceDown = bounceDown;
exports.bounceUp = bounceUp;
exports.bounceLeft = bounceLeft;
exports.bounceRight = bounceRight;