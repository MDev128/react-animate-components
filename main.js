/**
 * @author Nitin Tulswani <tulswani19@gmail.com>
 */

// Animation Component.
import bounceObj from './dist/animations/bounce/index';
import fadeObj from './dist/animations/fade/index';
import flipObj from './dist/animations/flip/index';
import lightObj from './dist/animations/lightspeed/index';
import rotateObj from './dist/animations/rotate/index';
import specialObj from './dist/animations/special';
import slideObj from './dist/animations/slide/index';
import PObj from './dist/animations/perspective/index';
import bingoObj from './dist/animations/Bingo/index';
import expanseObj from './dist/animations/expanse/index';

// Exposing keyframes for explicit use.
import {
	puffmeout,
	puffmein,
	vanishout,
	vanishin
} from './dist/animations/Bingo/styledBingo';

import {
  bounce,
  bounceDown,
  bounceUp,
  bounceLeft,
  bounceRight,
} from './dist/animations/bounce/styledBounce';

import { expanseUp, expanseDown, expanseLeft, expanseRight } from './dist/animations/expanse/styledExpanse';

import {
  down,
  downBig,
  up,
  upBig,
  left,
  right,
  leftBig,
  rightBig,
  fadeIn,
} from './dist/animations/fade/styledFade';

import {
	flip,
	flipX,
	flipY,
} from './dist/animations/flip/styledFlip';

import {
	lightIn,
	lightOut,
} from './dist/animations/lightspeed/styledLight';

import { perspectiveDown, perspectiveUp, perspectiveRight, perspectiveLeft } from './dist/animations/perspective/styledPerspective';

import { slideUp, slideDown, slideLeft, slideRight } from './dist/animations/slide/styledSlide';

import {
  rotateIn,
  rotateLeft,
  rotateRight,
  rotateUpRight,
  rotateUpLeft,
} from './dist/animations/rotate/styledRotate';

import {
  flash,
  rollOut,
  rollIn,
  rubber,
  swing,
  zoom,
  hinge,
  pulse,
  expandUp,
  entrance,
  hatch,
  starWars
} from './dist/animations/styledSpecial';

const { Bounce, BounceUp, BounceRight, BounceLeft, BounceDown } = bounceObj;

const {
  FadeIn,
  FadeInUp,
  FadeInRight,
  FadeInLeft,
  FadeInDown,
  FadeInUpBig,
  FadeInRightBig,
  FadeInLeftBig,
} = fadeObj;

const { Flip, FlipX, FlipY } = flipObj;

const { LightOut, LightIn } = lightObj;

const {
  RotateIn,
  RotateRight,
  RotateLeft,
  RotateUpRight,
  RotateUpLeft,
} = rotateObj;

const {
  Flash,
  RollOut,
  RollIn,
  Rubber,
  Swing,
  Zoom,
  Hinge,
  Pulse,
  ExpandUp,
  Entrance,
  Hatch,
  StarWars
} = specialObj;

const { SlideUp, SlideRight, SlideLeft, SlideDown } = slideObj;

const { PDown, PUp, PRight, PLeft } = PObj;

const { PuffOut, PuffIn, VanishOut, VanishIn } = bingoObj;

const { ExpanseUp, ExpanseRight, ExpanseDown, ExpanseLeft } = expanseObj;

/** keyframes */
export {
  puffmeout,
  puffmein,
  vanishout,
  vanishin,
  bounce,
  bounceDown,
  bounceUp,
  bounceLeft,
  bounceRight,
  expanseUp,
  expanseDown,
  expanseLeft,
  expanseRight,
  down,
  downBig,
  up,
  upBig,
  left,
  right,
  leftBig,
  rightBig,
  fadeIn,
  flip,
	flipX,
	flipY,
  lightIn,
	lightOut,
  perspectiveDown,
  perspectiveUp,
  perspectiveRight,
  perspectiveLeft,
  slideUp,
  slideDown,
  slideLeft,
  slideRight,
  rotateIn,
  rotateLeft,
  rotateRight,
  rotateUpRight,
  rotateUpLeft,
  flash,
  rollOut,
  rollIn,
  rubber,
  swing,
  zoom,
  hinge,
  pulse,
  expandUp,
  entrance,
  hatch,
  starWars
}

/** Animation components */
export {
  Bounce,
  BounceUp,
  BounceRight,
  BounceLeft,
  BounceDown,
  FadeIn,
  FadeInUp,
  FadeInRight,
  FadeInLeft,
  FadeInDown,
  FadeInUpBig,
  FadeInLeftBig,
  FadeInRightBig,
  Flip,
  FlipX,
  FlipY,
  LightOut,
  LightIn,
  RotateIn,
  RotateRight,
  RotateLeft,
  RotateUpRight,
  RotateUpLeft,
  Flash,
  RollOut,
  RollIn,
  Rubber,
  Swing,
  Zoom,
  Hinge,
  Pulse,
  SlideUp,
  SlideDown,
  SlideLeft,
  SlideRight,
  StarWars,
  ExpandUp,
  Entrance,
  Hatch,
  PDown,
  PUp,
  PRight,
  PLeft,
  PuffIn,
  PuffOut,
  VanishIn,
  VanishOut,
  ExpanseLeft,
  ExpanseRight,
  ExpanseDown,
  ExpanseUp
};
