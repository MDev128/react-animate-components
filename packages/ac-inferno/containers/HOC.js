// @flow

import Inferno from 'inferno';
import Component from 'inferno-component';
import { Render, derive } from '../utils/index';

type DefaultProps = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  as: string,
  component?: Function,
  children?: Element<any>,
  forceInterpolate?: Object,
  style?: Object
}

type Props = {
  duration: string,
  timingFunction: string,
  delay: string,
  direction: string,
  iterations: string,
  backfaceVisible: string,
  fillMode: string,
  playState: string,
  as: string,
  component?: Function,
  children?: Element<any>,
  forceInterpolate?: Object,
  style?: Object
}

type State = {
  styles: Object
}

function hoc(ComposedComponent: string, AnimationName: string): Function {
  class _Animation extends Component<DefaultProps, Props, State> {
    static displayName = `${ComposedComponent}`;

    static defaultProps = {
      duration: '1s',
      timingFunction: 'ease',
      delay: '0s',
      direction: 'normal',
      iterations: '1',
      backfaceVisible: 'visible',
      fillMode: 'none',
      playState: 'running',
      as: 'div'
    };

    state = {
      styles: {}
    };

    componentDidMount = () => {
      this.setAnimation(this.props);
    };

    componentWillReceiveProps = (nextProps: Props) => {
      // Interpolation of new animation properties
      const deriveInterpolationFromNextProps = derive(nextProps, AnimationName);

      // Old interpolation string
      const deriveInterpolationFromPrevProps = derive(
        this.props,
        AnimationName
      );

      if (
        deriveInterpolationFromNextProps !== deriveInterpolationFromPrevProps
      ) {
        this.setState({
          styles: Object.assign(
            {
              animation: `${deriveInterpolationFromNextProps}`,
              backfaceVisibility: `${nextProps.backfaceVisible}`
            },
            this.props.style || {}
          )
        });
      }
    };

    setAnimation = (props: Props) => {
      // Keyframes Interpolation and Force Interpolation
      const deriveInterpolation = derive(props, AnimationName);

      this.setState({
        styles: Object.assign(
          {
            animation: `${deriveInterpolation}`,
            backfaceVisibility: `${props.backfaceVisible}`
          },
          this.props.style || {}
        )
      });
    };

    render (): ?React$Element<any> {
      const {
        children,
        duration,
        timingFunction,
        delay,
        direction,
        iterations,
        backfaceVisible,
        fillMode,
        playState,
        forceInterpolate,
        as,
        style,
        component,
        ...rest
      } = this.props;

      return Render(ComposedComponent, this.props, this.state, rest, ComposedComponent);
    }
  }

  return _Animation;
}

export default hoc;
