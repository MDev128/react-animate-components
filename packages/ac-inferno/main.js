module.exports = require('./dist/ac-inferno.min.js');
