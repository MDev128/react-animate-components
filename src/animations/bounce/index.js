// @flow

import Bouncing from './styledBounce';
import HOC from '../../containers/HOC';
import type { Components } from '../../../types';

const bounceObj: Components = {
  Bounce: HOC('Bounce', Bouncing.bounce),
  BounceDown: HOC('BounceDown', Bouncing.bounceDown),
  BounceUp: HOC('BounceUp', Bouncing.bounceUp),
  BounceLeft: HOC('BounceLeft', Bouncing.bounceLeft),
  BounceRight: HOC('BounceRight', Bouncing.bounceRight),
};

export default bounceObj;
