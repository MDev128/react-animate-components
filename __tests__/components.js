import React from "react";

import renderer from "react-test-renderer";

import { keyframes } from 'styled-components';

import {shallow, mount} from 'enzyme';

import { Bounce } from "../main";

import { bounce } from '../src/animations/bounce/styledBounce';

import bounceObj from '../src/animations/bounce/index';

import * as All from '../main';

import HOC from '../src/animations/HOC';

const sample = keyframes`
	0% {
		opacity: 0;
	}

	100% {
		opacity: 1;
	}
`;

const Sample = HOC('Sample', sample);

it('keyframes should be string', () => {
	expect(typeof bounce).toBe('string');
});	

it('each component should export an object containing different animation components', () => {
	expect(typeof bounceObj).toBe('object');
	expect(Object.keys(bounceObj).length).not.toBe(0);
});

it('exports all the animation components', () => {
	expect(typeof All).toBe('object');
	expect(Object.keys(All).length).toBe(39);
});

it("renders the animation component", () => {
  const tree = renderer.create(<Bounce />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders the animation component with props", () => {
  const tree = renderer
    .create(
      <Bounce duration="2s" delay="4s" direction="reverse">
        <h1>Hello World!</h1>
      </Bounce>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('High Order Component should be a function', () => {
	expect(typeof HOC).toBe('function');
});

it('High Order Component should render an animation', () => {
	const tree = renderer.create(<Sample />).toJSON();

	expect(tree).toMatchSnapshot();
});

it('High Order Component should render the children with the props', () => {
	const tree = renderer
		.create(
			<Sample duration="2s">
				<h1>Hello World!</h1>
			</Sample>
		).toJSON();

	expect(tree).toMatchSnapshot();
});

it('renders the animation component with `block` props', () => {
	const tree = renderer.create(
		<Bounce block>
			<h1>Hello World!</h1>
		</Bounce>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

it('calls componentDidMount lifecycle method', () => {
	const wrapper = shallow(<Bounce />);
	wrapper.instance().componentDidMount();
});

it('calls renderRootWithBlock method when passed `block` as props', () => {
	const wrapper = shallow(<Bounce block />);
	wrapper.instance().renderRootWithBlock();
});

it('calls renderRootWithInline method (by default when no display props are passed)', () => {
	const wrapper = shallow(<Bounce />);
	wrapper.instance().renderRootWithInline();
});

it('component holds it state (defined) when no props are passed', () => {
	const wrapper = shallow(<Bounce />);

	expect(wrapper.state().styles).toBeDefined;
});

