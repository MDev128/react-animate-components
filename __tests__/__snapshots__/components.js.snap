// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`High Order Component should render an animation 1`] = `
<span
  style={
    Object {
      "animation": "
                UNeSH 1s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
/>
`;

exports[`High Order Component should render the children with the props 1`] = `
<span
  style={
    Object {
      "animation": "
                UNeSH 2s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</span>
`;

exports[`renders the animation component 1`] = `
<span
  style={
    Object {
      "animation": "
                bGJZDZ 1s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
/>
`;

exports[`renders the animation component with \`block\` props 1`] = `
<div
  style={
    Object {
      "animation": "
                bGJZDZ 1s ease 0s 1 normal
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "block",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</div>
`;

exports[`renders the animation component with props 1`] = `
<span
  style={
    Object {
      "animation": "
                bGJZDZ 2s ease 4s 1 reverse
                none running
              ",
      "backfaceVisibility": "visible",
      "display": "inline-block",
    }
  }
>
  <h1>
    Hello World!
  </h1>
</span>
`;
